src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -543,7 +556,25 @@ private async Task<ActionResult> PutFormData(string org, string app, Instance in
             // Trigger application business logic
             bool changedByCalculation = await _altinnApp.RunCalculation(serviceModel);
 
-            int instanceOwnerPartyId = int.Parse(instance.InstanceOwner.PartyId);
+            // Get old form data for comparison to identify changes in presentation texts
+            object oldData = await _dataService.GetFormData(
```

Comment:
Is this needed. Should you not compare with instance values?

Reply:
More I think about it we should only compare with values already on instance. Making the call to db would slow down the process.

Reply:
Updated implementation to not retrieve the old data